lib/cdo/secrets_config.rb

```diff
@@ -102,25 +103,22 @@ def stack_specific_secret_path
         @stack ? "CfnStack/#{@stack}/#{secret_key}" : nil
       end
 
-      EC2_METADATA_SERVICE_BASE_URL = URI('http://169.254.169.254/latest/meta-data/')
-
       # Get the CloudFormation Stack Name that the EC2 Instance this code is executing on belongs to.
       # @return [String]
       def self.current_stack_name
-        metadata_service_request = Net::HTTP.new(EC2_METADATA_SERVICE_BASE_URL.host, EC2_METADATA_SERVICE_BASE_URL.port)
-        # Set a short timeout so that when not executing on an EC2 Instance we fail fast.
-        metadata_service_request.open_timeout = metadata_service_request.read_timeout = 10
-        ec2_instance_id = metadata_service_request.request_get(EC2_METADATA_SERVICE_BASE_URL.path + 'instance-id').body
-        region = metadata_service_request.request_get(EC2_METADATA_SERVICE_BASE_URL.path + 'placement/region').body
+        ec2_instance_id = AWS::EC2.instance_id
+        return unless ec2_instance_id
+
+        region = AWS::EC2.region
+        return unless region
+
         ec2_client = Aws::EC2::Client.new(region: region)
         ec2_client.
           describe_tags({filters: [{name: "resource-id", values: [ec2_instance_id]}]}).
           tags.
           select {|tag| tag.key == 'aws:cloudformation:stack-name'}.
           first.
           value
-      rescue Net::OpenTimeout # This code is not executing on an AWS EC2 Instance nor in an ECS container or Lambda.
```

Comment:
This is a significant improvement of this logic. Thank you! We need this `current_stack_name` method to return `nil` when it executes on deployments of our application that are NOT in AWS (such as a local development environment or a Drone continuous integration build. Is that accomplished by the 2 short-circuit return statements? (`return unless ec2_instance_id` / `return unless region`)?

Reply:
The Secrets Config logic is fairly important. Maybe the new `AWS::EC2` module and this change should be released in a smaller Pull Request before the rest of this change?

Reply:
https://github.com/code-dot-org/code-dot-org/pull/55686